### PR TITLE
rewrite for brigade2 compatibility

### PR DESCRIPTION
**Pull request by krancour**
Complete rewrite of this GW for Brigade2 compatibility.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,94 +1,308 @@
 # Brigade Bitbucket Gateway
 
-Send [Bitbucket events](https://confluence.atlassian.com/bitbucket/manage-webhooks-735643732.html) into a [Brigade](https://github.com/brigadecore/brigade) pipeline. 
+This is a work-in-progress
+[Brigade 2](https://github.com/brigadecore/brigade/tree/v2)
+compatible gateway that receive events
```
Comment (vdice) [inline README.md]:
```suggestion
compatible gateway that receives events 
```

File: README.md [hunk 2]
```diff
@@ -1,94 +1,308 @@
...
+With this public IP in hand, edit your name servers and add an `A` record
+pointing your domain to the public IP.
+
+### 4. Create Webhooks
+
+In your browser, go to your Bitbucket repository for which you'd like to send
+webhooks to this gateway. From the menu on the left, select __Repository
+settings__ and then __Webhooks__. On this page, click __Add webhooks__.
+
+* In the __Tittle__ field, add a name for your webhook. It must be unique to this
```
Comment (vdice) [inline README.md]:
```suggestion
* In the __Title__ field, add a name for your webhook. It must be unique to this
```

File: README.md [hunk 3]
```diff
@@ -1,94 +1,308 @@
...
+
+Assuming this file were named `project.yaml`, you can create the project like
+so:
+
+```console
+$ brig project create --file project.yaml
 ```
 
-## Compatibility
+Opening an issue in the Bitbucket repo for which you configured webhooks to send
```
Comment (vdice) [inline README.md]:
```suggestion
Open an issue in the Bitbucket repo for which you configured webhooks to send
```

File: config_test.go [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+				_ restmachinery.APIClientOptions,
+				err error,
+			) {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), "value not found for")
+				require.Contains(t, err.Error(), "API_TOKEN")
+			},
+		},
+		{
+			name: "SUCCESS not set",
```
Comment (vdice) [inline config_test.go]: nit: perhaps just `"success"` as the name?

File: examples/project.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+          fooJob.primaryContainer.arguments = ["foo"];
+          await fooJob.run();
+
+          let barJob = new Job("bar", "debian:latest", event);
+          barJob.primaryContainer.command = ["echo"];
+          barJob.primaryContainer.arguments = ["bar"];
+          await barJob.run();
+        }
+
+        events.on("brigade.sh/github", "repo:forked", () => {
```
Comment (vdice) [inline examples/project.yaml]: Here and below we'd want `"brigade.sh/bitbucket"` as the source instead, right?

File: internal/webhooks/service.go [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+	) (core.EventList, error)
+}
+
+type service struct {
+	eventsClient core.EventsClient
+	config       ServiceConfig
+}
+
+// NewService returns an implementation of the Service interface for handling
+// (events) from Bitbucket.
```
Comment (vdice) [inline internal/webhooks/service.go]:
```suggestion
// webhooks (events) from Bitbucket.
```

File: config_test.go (hunk 4 above)
Comment (krancour) [reply config_test.go]: How do I keep doing that?

File: examples/project.yaml (hunk 5 above)
Comment (krancour) [reply examples/project.yaml]: Nice catch.

**Comment by krancour** [comment]
@vdice all suggested changes applied.

**Review by vdice — APPROVED** [review]
LGTM!

**Pull request merged**